mlflow/entities/param.py

```diff
@@ -1,21 +1,17 @@
-import sys
-
 from mlflow.entities._mlflow_object import _MLflowObject
 from mlflow.protos.service_pb2 import Param as ProtoParam
 
+from .conversion_utils import convert_to_str_if_possible
 
 class Param(_MLflowObject):
     """
     Parameter object.
     """
 
     def __init__(self, key, value):
-        if "pyspark.ml" in sys.modules:
-            import pyspark.ml.param
+        key = convert_to_str_if_possible(key)
+        value = convert_to_str_if_possible(value)
```

Comment:
I would like to keep `if isinstance(key, pyspark.ml.param.Param):` and `key = key.name` logic inside `Param.__init__`, because convert_to_str_if_possible is a generic function but the `Param.name` logic only apply to "Param key" case.